packages/wrangler/src/user/user.ts

```diff
@@ -392,6 +392,15 @@ export function validateScopeKeys(
 }
 
 function getCallbackUrl(host = "localhost", port = 8976) {
+	if (
+		host === "localhost" ||
+		host === "0.0.0.0" ||
+		host === "::" ||
+		host === "::1" ||
+		host === "127.0.0.1"
+	) {
+		return `http://localhost:${port}/oauth/callback`;
+	}
 	return `http://${host}:${port}/oauth/callback`;
```

Comment:
I have a feeling that we should always return `localhost` for this URL.

The only URIs that are allowed are configured in the OAuth provider, and I can't find where we (Cloudflare) control that configuration.

If we were to do a check here it should be based on an allow-list rather than a block-list as is defined here.